From da0cf4105dea7b5701fdbcce68421a2251f1abb6 Mon Sep 17 00:00:00 2001
From: Shawn Landen <shawnlandden@gmail.com>
Date: Fri, 3 Aug 2012 02:28:28 +0000
Subject: [PATCH] logs-show: fix off-by-one error

Ellipsize lines that are one character too long.
(cherry picked from commit 193556b69e982fd37f88209c26a5783245cf2437)

Conflicts:
	src/shared/logs-show.c
---
 src/shared/logs-show.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/logs-show.c b/src/shared/logs-show.c
index 2219c80..9c6dcbf 100644
--- a/src/shared/logs-show.c
+++ b/src/shared/logs-show.c
@@ -230,7 +230,7 @@ static int output_short(sd_journal *j, unsigned line, unsigned n_columns, bool s
         else if (contains_unprintable(message, message_len)) {
                 char bytes[FORMAT_BYTES_MAX];
                 printf(": [%s blob data]\n", format_bytes(bytes, sizeof(bytes), message_len));
-        } else if (message_len + n < n_columns)
+        } else if (message_len + n + 1 < n_columns)
                 printf(": %.*s\n", (int) message_len, message);
         else if (n < n_columns && n_columns - n - 2 >= 3) {
                 char *e;
